### ISSUE TITLE
testnet support is broken

### ISSUE THREAD

**Issue by greenaddress** [esotericnonsense/bitcoind-ncurses#18]
Hi

I use bitcoind-ncurses regularly against mainnet. Thank you for making bitcoind-ncurses available!

Today I tried to use it against testnet and it failed because of the block_subsidy.

I added a quick hack fix to my local repo and it seems to work ok but obviously it is not as nice as a real fix as per the suggestion in the TODO ("# this will fail after block 420,000. TODO: stop being lazy and do it properly")

See the fix below, hope it helps others.

```diff
diff --git a/monitor.py b/monitor.py
index 5a3e3e1..a075630 100644
--- a/monitor.py
+++ b/monitor.py
@@ -3,6 +3,7 @@ import curses, time, math
 
 import global_mod as g
 import footer
+from decimal import Decimal
 
 def draw_window(state, old_window):
     # TODO: only draw parts that actually changed
@@ -55,10 +56,12 @@ def draw_window(state, old_window):
                     block_subsidy = 50
                 elif state['mininginfo']['blocks'] < 420000:
                     block_subsidy = 25
+                elif state['mininginfo']['blocks'] < 600000:
+                    block_subsidy = 12.5
 
                 if block_subsidy: # this will fail after block 420,000. TODO: stop being lazy and do it properly
                     coinbase_amount = blockdata['coinbase_amount']
-                    total_fees = coinbase_amount - block_subsidy # assumption, mostly correct
+                    total_fees = coinbase_amount - Decimal(block_subsidy) # assumption, mostly correct
 
                     if coinbase_amount > 0:
                         fee_percentage = "%0.2f" % ((total_fees / coinbase_amount) * 100)
```

**Comment by esotericnonsense** [comment]
It's also a ticking timebomb waiting to break when the block reward halves.
I don't recall why the actual algorithm wasn't used, to be honest.

**Comment by esotericnonsense** [comment]
This should be fixed as of f372d260. I have tested it on live but not testnet. Any issues, please re-open.

block_subsidy = 50 * 0.5**(B/210000) where B = block number and we use integer division.

**Comment by greenaddress** [comment]
thanks!
